R/rows-d.R

```diff
@@ -0,0 +1,214 @@
+#' Persisting data
+#'
+#' @description
+#' \lifecycle{experimental}
+#'
+#' These methods provide a framework for manipulating individual rows
+#' in existing tables, modeled after the SQL operations
+#' `INSERT`, `UPDATE` and `DELETE`.
+#' All operations expect existing and new data to be compatible.
+#'
+#' @param .data Target table object.
+#' @param .key Key columns, unquoted.
+#'   The default varies for the different operations:
+#'   - For `rows_insert()`: no keys.
+#'   - For `rows_update()`, `rows_patch()` and `rows_upsert()`: first column.
+#'   - For `rows_delete()`: all columns.
+#' @param ... New data.
+#'   If unnamed, must be a single object of the same class as `.data`.
+#'   If named, will be passed on to [tibble()].
+#'
+#' @return A tbl object of the same structure as `.data`.
+#'
+#' @name rows
+#' @example example/rows-d.R
+NULL
+
+
+#' rows_insert
+#'
+#' `rows_insert()` adds new rows.
+#' This operation corresponds to `INSERT` in SQL.
+#' If `.key` is given, no two rows with the same values in the key columns
+#' are permitted.
+#' @param .sort Should the output be sorted by the key columns?
+#' @rdname rows
+#' @export
+rows_insert <- function(.data, ..., .key = NULL) {
```

Comment:
I now think the basic interface should be `function(x, y, key = NULL)` to match the joins. I think it's unnecessary to allow free floating columns, and it makes the interface substantially simpler. 

That also means that we can be more clear that the output of these functions are consistent with the output of the joins: the return type is always the same type as x.